extensions/amp-form/0.1/amp-form.js

```diff
@@ -797,19 +798,16 @@ export class AmpForm {
    */
   handleSsrTemplateResponse_(response, trust) {
     const init = response['init'];
+    // response['body'] is serialized as a string in the response.
+    const body = response['body'] ? parseJson(response['body']) : null;
```

Comment:
Is `response.body` guaranteed to be well-formed JSON? I.e. is it ok if `parseJson` throws a runtime error?

If not, use `tryParseJson`.

Reply:
`response.body` is serialized as a string here, so it should be reasonable to assume that it is well-formed:
https://github.com/ampproject/amphtml/blob/a0f855807981b65fa662395e1ae30f59c68be3fa/src/utils/xhr-utils.js#L174

If `parseJson` fails in practice, wouldn't we want a runtime error to surface this rather than fail silently anyway?

Reply:
Yes, but would the runtime error in this method cause the form's `[submit-success]` or `[submit-error]` template not to render (vs. just sending null data to the event handler)?

Reply:
Good point. To both continue down the code path _and_ surface the JSON parse issue, let's use `tryParseJson` with a callback to `user().warn` with any error.

Reply:
I'd use `user().error` for malformed data. Note that it doesn't cause a runtime exception.